File: frontend/src/types/document.ts
// ─── 文档实体 ──────────────────────────────────────────────

export interface Document {
  id: number
  title: string
  docType: string
  minioKey?: string
  fileSizeBytes?: number
  fileHash?: string
  version: number
  status: string
  approvedBy?: number
  approvedAt?: string
  expiresAt?: string
  createdAt: string
  updatedAt: string
  createdBy?: string
}

export interface DocumentVersion {
  id: number
  documentId: number
  version: number
  minioKey?: string
  changeDescription?: string
  createdBy?: string
  createdAt: string
}

// ─── DTO ────────────────────────────────────────────────────

export interface CreateDocument {
  title: string
  docType: string
  minioKey?: string
  fileSizeBytes?: number
  fileHash?: string
}

export interface UploadResponse {
  minioKey: string
  size: number
  hash: string
}

export interface ApproveDocument {
  approvedBy: number
}

export interface RejectDocument {
  reason: string
}

// ─── 常量（下拉选项） ──────────────────────────────────────

export const DOC_TYPE_OPTIONS = [
  { value: 'sop', label: 'SOP' },
  { value: 'work_instruction', label: '作业指导书' },
  { value: 'inspection_standard', label: '检验标准' },
  { value: '8d_report', label: '8D报告' },
  { value: 'audit_report', label: '审核报告' },
  { value: 'other', label: '其他' },
]

export const DOC_TYPE_MAP: Record<string, string> = {
  sop: 'SOP',
  work_instruction: '作业指导书',
  inspection_standard: '检验标准',
  '8d_report': '8D报告',
  audit_report: '审核报告',
  other: '其他',
}

export const DOC_STATUS_OPTIONS = [
  { value: 'draft', label: '草稿', type: 'info' },
  { value: 'reviewing', label: '审批中', type: 'warning' },
  { value: 'approved', label: '已批准', type: 'success' },
  { value: 'archived', label: '已归档', type: '' },
]

export const DOC_STATUS_MAP: Record<string, { label: string; type: '' | 'info' | 'warning' | 'success' }> = {
  draft: { label: '草稿', type: 'info' },
  reviewing: { label: '审批中', type: 'warning' },
  approved: { label: '已批准', type: 'success' },
  archived: { label: '已归档', type: '' },
}

File: backend/Models/M12/Document.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QM_AI.API.Models.M12;

/// <summary>
/// 文件/文档
/// </summary>
[Table("documents")]
public class Document
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    /// <summary>文件标题</summary>
    [Required]
    [MaxLength(500)]
    public string Title { get; set; } = string.Empty;

    /// <summary>文档类型：sop / work_instruction / inspection_standard / 8d_report / audit_report / other</summary>
    [Required]
    [MaxLength(20)]
    public string DocType { get; set; } = string.Empty;

    /// <summary>MinIO 对象键</summary>
    [Required]
    [MaxLength(500)]
    public string MinioKey { get; set; } = string.Empty;

    /// <summary>文件大小（字节）</summary>
    public long? FileSizeBytes { get; set; }

    /// <summary>SHA-256 哈希</summary>
    [MaxLength(64)]
    public string? FileHash { get; set; }

    /// <summary>版本号</summary>
    public int Version { get; set; } = 1;

    /// <summary>状态：draft / reviewing / approved / archived</summary>
    [Required]
    [MaxLength(10)]
    public string Status { get; set; } = "draft";

    /// <summary>审批人ID</summary>
    public long? ApprovedBy { get; set; }

    /// <summary>审批人标识（字符串，供服务层兼容）</summary>
    public string? ApprovedByStr { get; set; }

    /// <summary>驳回理由</summary>
    public string? RejectionReason { get; set; }

    /// <summary>审批时间</summary>
    public DateTime? ApprovedAt { get; set; }

    /// <summary>有效期</summary>
    public DateOnly? ExpiresAt { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    /// <summary>创建人</summary>
    public long CreatedBy { get; set; }

    // Navigation
    public virtual ICollection<DocumentVersion>? Versions { get; set; } = new List<DocumentVersion>();
}
